mock/py/mockbuild/podman.py

```diff
@@ -114,21 +116,33 @@ def mounted_image(self):
             subprocess.run(cmd_umount, stdout=subprocess.PIPE,
                            stderr=subprocess.PIPE, check=True)
 
-    def get_image_digest(self):
+    def get_layers_digest(self):
         """
-        Get the "sha256:..." string for the image we work with.
+        Get sha256 digest of RootFS layers. This must be identical for
+        all images containing same order of layers, thus it can be used
+        as the check that we've loaded same image.
         """
         check = [self.podman_binary, "image", "inspect", self.image,
-                 "--format", "{{ .Digest }}"]
+                 "--format", "{{ .RootFS }}"]
         result = subprocess.run(check, stdout=subprocess.PIPE,
                                 stderr=subprocess.PIPE, check=False,
                                 encoding="utf8")
         if result.returncode:
             raise BootstrapError(f"Can't get {self.image} podman image digest: {result.stderr}")
         result = result.stdout.strip()
-        if len(result.splitlines()) != 1:
-            raise BootstrapError(f"The digest of {self.image} image is not a single-line string")
-        return result
+        try:
+            rootfs = json.loads(result)
+            layers = rootfs['layers']
+        except json.JSONDecodeError as exc:
+            raise BootstrapError(f"The RootFS layers of {self.image} "
+                                  "are not json-formatted.") from exc
+        except (KeyError, TypeError) as exc:
+            raise BootstrapError(f"The Layers of {self.image} "
+                                  "are not in expected format.") from exc
+        sha = hashlib.sha256()
+        for layer in layers:
+            sha.update(layer.encode()) # as it was decoded from json, it is unicode again
+        return sha.hexdigest()
```

Comment:
This doesn't give us a JSON formatted output:
```
$ podman image inspect -f=json --format '{{ .RootFS }}' registry.access.redhat.com/ubi8/ubi 
{layers [sha256:3200b171908c5b74d4e1781a79a0fc729b13c29fc3ce70687384c8cc349f509d]}
```

The `podman image inspect` output seems to be JSON formatted by default, unless you provide the option `--format`.

Reply:
```
diff --git a/mock/py/mockbuild/podman.py b/mock/py/mockbuild/podman.py
index 1a1063cf..6287013e 100644
--- a/mock/py/mockbuild/podman.py
+++ b/mock/py/mockbuild/podman.py
@@ -122,20 +122,19 @@ class Podman:
         all images containing same order of layers, thus it can be used
         as the check that we've loaded same image.
         """
-        check = [self.podman_binary, "image", "inspect", self.image,
-                 "--format", "{{ .RootFS }}"]
+        check = [self.podman_binary, "image", "inspect", self.image]
         result = subprocess.run(check, stdout=subprocess.PIPE,
                                 stderr=subprocess.PIPE, check=False,
                                 encoding="utf8")
         if result.returncode:
-            raise BootstrapError(f"Can't get {self.image} podman image digest: {result.stderr}")
+            raise BootstrapError(f"Can't inspect {self.image} podman image: {result.stderr}")
         result = result.stdout.strip()
         try:
-            rootfs = json.loads(result)
-            layers = rootfs['layers']
+            inspect_data = json.loads(result)
+            layers = inspect_data[0]['RootFS']['Layers']
         except json.JSONDecodeError as exc:
-            raise BootstrapError(f"The RootFS layers of {self.image} "
-                                  "are not json-formatted.") from exc
+            raise BootstrapError(f"The inspect output of {self.image} "
+                                  "is not a json-formatted string.") from exc
         except (KeyError, TypeError) as exc:
             raise BootstrapError(f"The Layers of {self.image} "
                                   "are not in expected format.") from exc